### PR TITLE
mfhdf/test/tdatainfo.c: General cleanup

### PR DESCRIPTION
**Pull request by schwehr**
Demonstrate initial clean up of one test file

- include what you use (iwyu) but then some hand editing
- remove signatures for comments
- remove definition of DATAINFO_TESTER
- remove redundant test function prototypes
- convert `#define`s to const int and const char *
- localize variables / combine definition and initialization
- add const
- use stdint.h: int16 and int32 -> int16_t and int32_t
- intn -> int
- add static on more local functions
- remove empty statement: `;`
- remove `#if 0` block of code
- remove commented out checks

### CODE REVIEW

File: mfhdf/test/tdatainfo.c [hunk 1]
```diff
@@ -943,113 +777,102 @@ static intn test_compressed_SDSs()
  * number of blocks for data, which should be 0.
  */
 /****************************************************************************
- Name: test_empty_SDSs() - tests special but empty SDSs
+ Name: test_chunked_empty_SDSs() - tests special but empty SDSs
```
Comment (bmribler) [inline mfhdf/test/tdatainfo.c:780]: The term "chunked" is incorrect in this test function.

File: mfhdf/test/tdatainfo.c [hunk 2]
```diff
@@ -1207,35 +1003,21 @@ static intn test_chunked_partial()
     /* Record number of values the SDS can have */
     sds_info.n_values = 1 * 10; /* chunk has 1 dim of size 10 */
 
-#if 0 /* need to figure out how to test this feature */
-    status = SDgetdatainfo(sds_id, NULL, 0, info_count, sds_info.offsets, sds_info.lengths);
-    CHECK(status, FAIL, "test_chunked_partial: SDgetdatainfo");
-#endif
-
```
Comment (bmribler) [inline mfhdf/test/tdatainfo.c:1214]: This feature should be tested instead of removed.

File: mfhdf/test/tdatainfo.c [hunk 3]
```diff
@@ -1755,28 +1505,14 @@ static intn test_extend_SDSs()
...
     TESTING("getting location info of data (tdatainfo.c)");
 
-    /* Test nonspecial SDSs */
-    num_errs = num_errs + test_nonspecial_SDSs();
-
-    /* Test compressed SDSs */
+    int num_errs = test_nonspecial_SDSs();
     num_errs = num_errs + test_compressed_SDSs();
-
-    /* Test chunked empty SDSs */
```
Comment (bmribler) [inline mfhdf/test/tdatainfo.c:1770]: Comment from the original code was incorrect.  Should be "Test special empty SDSs".

**Comment by schwehr** [comment]
Thanks for the feedback.  I'm probably not the right person to be doing the fixups an tdatainfo.c.  Also a lot of cleanup has already happened.  I'm going to close this PR.  Feel free to reopen if you'd like me to give this file a go.

**Comment by bmribler** [comment]
OK, we will see what are left here after all the cleanups are done.  Thank you!!

**Pull request closed without merging**